### PR TITLE
Throw error main

### LINKED ISSUE #21: Throw error on unrecognized command-line argument
Currently, it just skips arguments not matched in the if-else block. The expected behaviour should be that `chemilang -O -Y example.chem` should throw an error.

### PR DESCRIPTION
**Pull request by Andrikz**
Don't know if this is the way to do this or not?

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -16,20 +16,30 @@ int main(int argc, char *argv[]) {
...
+		}
+		if (drv.parse_file(filename) == 0) {
+			frontend.drv = &drv;
+			frontend.WriteFile();
+		} else {
+			std::cout << "Compilation error" << std::endl;
+			Frontend::Helper();
+		}
+		return 0;
+	} catch (std::invalid_argument &ex) {
```
Comment (RasmusRendal) [inline src/main.cpp]: Jeg tror ikke at try-catches er den rigtige måde at håndtere det her på. At wrappe alt koden i en try-catch, især dele af koden der ikke kan kaste den exception, er ret grimt, når man bare kunne skrive print i stedet for `throw`. 

Desuden, nu hvor vi er i gang med at fixe `main`, burde vi smide et `return 1` til sidst. Derefter kan vi explicit returne 0 hvis at vores compilation faktisk lykkedes. Lige nu returner vi f.eks. stadig 0 under en compilation error, hvilket jo er helt galt.

File: src/frontend.cpp [hunk 2]
```diff
@@ -12,13 +12,21 @@ void Frontend::WriteFile() {
...
 	if (errorCode == fileError) {
 		std::cout << "Error: No file for parsing" << std::endl;
+	} else if (errorCode == argError) {
+		std::cout << "Invalid command-line argument used " << input << '\n';
+		std::cout << "Please see -h for more info" << std::endl;
+	}
+}
+
+void Frontend::Helper(Error errorCode) {
+	if (errorCode == helpArgument) {
```
Comment (RasmusRendal) [inline src/frontend.cpp]: Maybe just remove this if statement, and rename the function to `PrintHelp`?

File: src/main.cpp [hunk 3]
```diff
@@ -16,21 +16,27 @@ int main(int argc, char *argv[]) {
...
 		frontend.drv = &drv;
 		frontend.WriteFile();
 	} else {
 		std::cout << "Compilation error" << std::endl;
 	}
-	Frontend::Helper();
-
+	
+        Frontend::Helper();
 	return 0;
```
Comment (RasmusRendal) [inline src/main.cpp:36]: If it gets to the end, without returning 0 beforehand, it should probably return 1.

**Review by RasmusRendal — CHANGES_REQUESTED** [review]

File: src/main.cpp (hunk 3 above)
Comment (Andrikz) [reply src/main.cpp:36]: I just fixed it.

File: src/frontend.cpp (hunk 2 above)
Comment (Andrikz) [reply src/frontend.cpp]: If we need to make more helpers for different arguments in the command-line the we can put them in the helper method.

Comment (RasmusRendal) [reply src/frontend.cpp]: Well, but until then, let's not overcomplicate things

File: src/main.cpp [hunk 4]
```diff
@@ -12,25 +12,25 @@ int main(int argc, char *argv[]) {
 	std::string filename;
 	driver drv;
 	if (argv[argc - 1] == std::string("-h")) {
-		Frontend::Helper(helpArgument);
-		return 0;
+		Frontend::PrintHelper();
+		return 1;
```
Comment (RasmusRendal) [inline src/main.cpp]: You're not supposed to return 1 when you print help after getting asked for it. Try running `grep --help` for example.

File: src/frontend.cpp [hunk 5]
```diff
@@ -12,10 +12,17 @@ void Frontend::WriteFile() {
 	std::cout << "Output written to " << outputFileName << std::endl;
 }
 
-void Frontend::Helper(Error errorCode) {
+void Frontend::Exception(Error errorCode, const std::string &input) {
 	if (errorCode == fileError) {
-		std::cout << "Error: No file for parsing" << std::endl;
+		std::cout << "Error: No file for parsing" << input << std::endl;
 	}
+	if (errorCode == argError) {
```
Comment (RasmusRendal) [inline src/frontend.cpp]: This could be a switch case, for clarity's sake

File: src/main.cpp [hunk 6]
```diff
@@ -12,25 +12,25 @@ int main(int argc, char *argv[]) {
...
 			frontend.outputFileName = std::string(argv[++i]);
+		} else {
+			Frontend::Exception(argError, argv[i]);
+			return 1;
 		}
 	}
-
 	if (drv.parse_file(filename) == 0) {
 		frontend.drv = &drv;
 		frontend.WriteFile();
```
Comment (RasmusRendal) [inline src/main.cpp:31]: It should probably not end up returning 1 when we successfully parse the file.

**Review by RasmusRendal — CHANGES_REQUESTED** [review]

File: src/main.cpp (hunk 6 above)
Comment (Andrikz) [reply src/main.cpp:31]: What do you mean?

Comment (RasmusRendal) [reply src/main.cpp:31]: Think about the control flow. If `drv.parse_file(filename)` is 0, it writes ot a file, and exits the if-else block. Then it goes down into the `return 1` on line 35, acting as if executing the binary returned an error, when it actually went fine

**Review by RasmusRendal — APPROVED** [review]

**Pull request merged**